rdt/transformers/base.py

```diff
@@ -9,20 +9,118 @@ class BaseTransformer:
     and ``fit_transform`` method is already implemented.
     """
 
-    def fit(self, data):
+    INPUT_TYPE = None
+    OUTPUT_TYPES = None
+    DETERMINISTIC_TRANSFORM = None
+    DETERMINISTIC_REVERSE = None
+    COMPOSITION_IS_IDENTITY = None
+    NEXT_TRANSFORMERS = None
+
+    def get_input_type(self):
+        """Return the input type supported by the transformer.
+
+        Returns:
+            string:
+                Accepted input type of the transformer.
+        """
+        return self.INPUT_TYPE
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        return self.OUTPUT_TYPES
+
+    def is_transform_deterministic(self):
+        """Return whether the transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the transform is deterministic.
+        """
+        return self.DETERMINISTIC_TRANSFORM
+
+    def is_reverse_deterministic(self):
+        """Return whether the reverse transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the reverse transform is deterministic.
+        """
+        return self.DETERMINISTIC_REVERSE
+
+    def is_composition_identity(self):
+        """Return whether composition of transform and reverse transform produces the input data.
+
+        Returns:
+            bool:
+                Whether or not transforming and then reverse transforming returns the input data.
+        """
+        return self.COMPOSITION_IS_IDENTITY
+
+    def get_next_transformers(self):
+        """Return the suggested next transformer to be used for each column.
+
+        Returns:
+            dict:
+                Mapping from transformed column names to the transformers to apply to each column.
+        """
+        output_types = self.get_output_types()
+        output_columns = list(output_types.keys())
+        columns_next_transformers = {column: None for column in output_columns}
+
+        if self.NEXT_TRANSFORMERS:
+            return {**self.NEXT_TRANSFORMERS, **columns_next_transformers}
+
+        return columns_next_transformers
+
+    def fit(self, data, columns):
         """Fit the transformer to the data.
 
         Args:
             data (pandas.Series or numpy.array):
                 Data to transform.
+            columns (list):
+                List of column names from the data.
+        """
+        self._columns = columns
```

Comment:
Here we should also craft a unique identifier of the columns by concatenating the names of the columns with a separator and storing it as `self._column_name`.

For example, if `columns == ['year', 'month', 'day']`, `self._column_name == 'year#month#day'`. We just need to make sure that the separator we use does not end up generating a column that already exists in the input table.

Reply:
If there is only one column, should the separator still be appended?

Reply:
If we go with @csala suggestion regarding the `get_output_types()`, I think we will always have to append something, yes.